static/usr/lib/core/run-snapd-from-snap

```diff
@@ -5,31 +5,40 @@
 
 set -eux
 
-# run_on_unseeded will mount/run snapd on an unseeded system
+# run_on_unseeded will run snapd on an unseeded system. The snapd snap
+# is expected to have been mounted and current link is expected to
+# have been created by the initramfs.
 run_on_unseeded() {
-    SNAPD_BASE_DIR="/run/mnt/snapd"
-
-    # We need to initialize /snap/snapd/current symlink so that the
-    # dynanic linker
-    # /snap/snapd/current/usr/lib/x86_64-linux-gnu/ld-linux-x86-64.so.2
-    # is available to run snapd.
-    [ -d /snap/snapd ] || mkdir -p /snap/snapd
-    ln -sf "${SNAPD_BASE_DIR}" /snap/snapd/current
+    SNAPD_BASE_DIR="/snap/snapd/current"
+    if ! mountpoint -q "$SNAPD_BASE_DIR"; then
+        # Compatibility with old way where we had a duplicated mount in /run
+        # and "current" link was not created by initramfs.
+        SNAPD_BASE_DIR="/run/mnt/snapd"
+        # We need to initialize /snap/snapd/current symlink so that the
+        # dynamic linker
+        # /snap/snapd/current/usr/lib/x86_64-linux-gnu/ld-linux-x86-64.so.2
+        # is available to run snapd.
+        [ -d /snap/snapd ] || mkdir -p /snap/snapd
```

Comment:
Think about it, the condition was useless here, right? Maybe you can remove it in the same time.